File: 0012-integer-to-roman/0012-integer-to-roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
  let romans = [
    ["M", 1000],
    ["CM", 900],
    ["D", 500],
    ["CD", 400],
    ["C", 100],
    ["XC", 90],
    ["L", 50],
    ["XL", 40],
    ["X", 10],
    ["IX", 9],
    ["V", 5],
    ["IV", 4],
    ["I", 1]
  ];
  let ans = "";
  let res = num;

  while (res != 0){
    for (let i = 0; i<romans.length; i++){
      if (res >= romans[i][1]){
        ans = ans.concat('',romans[i][0]);
        res -= romans[i][1];
        break;
      }
    }
  }
  return ans;
};

File: 0012-integer-to-roman/0012-integer-to-roman.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        roman = [
            ("I", 1),
            ("IV", 4),
            ("V", 5),
            ("IX", 9),
            ("X", 10),
            ("XL", 40),
            ("L", 50),
            ("XC", 90),
            ("C", 100),
            ("CD", 400),
            ("D", 500),
            ("CM", 900),
            ("M", 1000)
        ]

        res = ""
        for i in range(len(roman)-1, -1, -1):
            while num >= roman[i][1]:
                res += roman[i][0]
                num -= roman[i][1]

        return res